### PR TITLE
First draft of support library

### PR DESCRIPTION
**Pull request by barulicm**
Implements CircuitPython support for Gizmo.

* Adds the `circuitpython_gizmo` module, which provides the `Gizmo` object as the primary interface for student code.
* Adds a handful of examples to show how to use the Gizmo support library along with common Adafruit libraries to do basic robot tasks.
* Fills out the readme.

### CODE REVIEW

File: circuitpython_gizmo.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            self.i2c.unlock()
+
+    def refresh(self) -> None:
+        """Polls system processor for latest gamepad state"""
+        if self.i2c is None:
+            return
+        buffer = bytearray(18)
+        try:
+            self.i2c.readfrom_into(8, buffer)
+            self.axes = GizmoAxes(buffer[0:6])
```
Comment (the-maldridge) [inline circuitpython_gizmo.py]: Does this not cause object churn in memory?

**Review by the-maldridge — CHANGES_REQUESTED** [review]
One question for memory management, and then on git formatting this PR needs to be squashed to one commit and match the format of the rest of the library, namely it needs to be of the following format:

```
path: thing you changed
```

So for this commit, a suitable commit could be

```
all: initial implementation
```

File: circuitpython_gizmo.py (hunk 1 above)
Comment (barulicm) [reply circuitpython_gizmo.py]: It does. I've refactored to not have it create new objects in `refresh`.

**Comment by barulicm** [comment]
Do you prefer to force push squashed commits to PRs as changes or made or "squash and merge" when the PRs ready?

**Comment by the-maldridge** [comment]
The merge will be a fast forward, so you should make the branch correct prior to merge. (yes, force push)

File: circuitpython_gizmo.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    def deinit(self):
+        """Releases lock on I2C device"""
+        if self.i2c is not None:
+            self.i2c.unlock()
+
+    def refresh(self) -> None:
+        """Polls system processor for latest gamepad state"""
+        if self.i2c is None:
+            return
+        buffer = bytearray(18)
```
Comment (the-maldridge) [inline circuitpython_gizmo.py:136]: I'm fairly confident this churns memory, but it can be addressed in a later patch.

**Pull request merged**